apps/explorer/src/pages/checkpoints/CheckpointDetail.tsx

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useFeature, useGrowthBook } from '@growthbook/growthbook-react';
+import { useQuery } from '@tanstack/react-query';
+import { Navigate, useParams } from 'react-router-dom';
+
+import { CheckpointTransactions } from './Transactions';
+
+import { useRpc } from '~/hooks/useRpc';
+import { Banner } from '~/ui/Banner';
+import { DescriptionList, DescriptionItem } from '~/ui/DescriptionList';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { PageHeader } from '~/ui/PageHeader';
+import { Tab, TabGroup, TabList, TabPanel, TabPanels } from '~/ui/Tabs';
+import { Text } from '~/ui/Text';
+import { GROWTHBOOK_FEATURES } from '~/utils/growthbook';
+import { convertNumberToDate } from '~/utils/timeUtils';
+
+function CheckpointDetail() {
+    const { digest } = useParams<{ digest: string }>();
+    const rpc = useRpc();
+
+    const checkpointQuery = useQuery(['checkpoints', digest], () =>
+        rpc.getCheckpoint(digest!)
+    );
+
+    // todo: add user_signatures to combined `getCheckpoint` endpoint
+    const contentsQuery = useQuery(
+        ['contents'],
+        () => rpc.getCheckpointContents(checkpoint.sequenceNumber),
+        { enabled: !!checkpointQuery.data }
+    );
+
+    if (checkpointQuery.isError)
+        return (
+            <Banner variant="error" fullWidth>
+                There was an issue retrieving data for checkpoint: {digest}
+            </Banner>
+        );
+
+    if (checkpointQuery.isLoading) return <LoadingSpinner />;
+
+    const {
+        data: { epochRollingGasCostSummary, ...checkpoint },
+    } = checkpointQuery;
+
+    return (
+        <div className="flex flex-col space-y-12">
+            <PageHeader title={checkpoint.digest} type="Checkpoint" />
+            <div className="space-y-8">
+                <TabGroup as="div" size="lg">
```

Comment:
I wonder if we should default `as="div"` in our wrapper. Not something you need to do just thinking aloud.